tests/e2e/p/staking_rewards.go

```diff
@@ -0,0 +1,284 @@
+					Threshold: 1,
+					Addrs:     []ids.ShortID{alphaValidationRewardKey.Address()},
+				},
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{alphaDelegationRewardKey.Address()},
+				},
+				delegationFee,
+			)
+			require.NoError(err)
+		})
+
+		betaValidatorStartTime := time.Now().Add(validatorStartTimeDiff)
+		betaValidatorEndTime := betaValidatorStartTime.Add(validationPeriod)
+		tests.Outf("beta node validation period starting at: %v\n", betaValidatorStartTime)
+
+		ginkgo.By("adding beta node as a validator", func() {
+			_, err := pWallet.IssueAddPermissionlessValidatorTx(
+				&txs.SubnetValidator{Validator: txs.Validator{
+					NodeID: betaNodeID,
+					Start:  uint64(betaValidatorStartTime.Unix()),
+					End:    uint64(betaValidatorEndTime.Unix()),
+					Wght:   weight,
+				}},
+				betaPOP,
+				pWallet.AVAXAssetID(),
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{betaValidationRewardKey.Address()},
+				},
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{betaDelegationRewardKey.Address()},
+				},
+				delegationFee,
+			)
+			require.NoError(err)
+		})
+
+		gammaDelegatorStartTime := time.Now().Add(validatorStartTimeDiff)
+		tests.Outf("gamma delegation period starting at: %v\n", gammaDelegatorStartTime)
+
+		ginkgo.By("adding gamma as delegator to the alpha node", func() {
+			_, err := pWallet.IssueAddPermissionlessDelegatorTx(
+				&txs.SubnetValidator{Validator: txs.Validator{
+					NodeID: alphaNodeID,
+					Start:  uint64(gammaDelegatorStartTime.Unix()),
+					End:    uint64(gammaDelegatorStartTime.Add(delegationPeriod).Unix()),
+					Wght:   weight,
+				}},
+				pWallet.AVAXAssetID(),
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{gammaDelegationRewardKey.Address()},
+				},
+			)
+			require.NoError(err)
+		})
+
+		deltaDelegatorStartTime := time.Now().Add(validatorStartTimeDiff)
+		tests.Outf("delta delegation period starting at: %v\n", deltaDelegatorStartTime)
+
+		ginkgo.By("adding delta as delegator to the beta node", func() {
+			_, err := pWallet.IssueAddPermissionlessDelegatorTx(
+				&txs.SubnetValidator{Validator: txs.Validator{
+					NodeID: betaNodeID,
+					Start:  uint64(deltaDelegatorStartTime.Unix()),
+					End:    uint64(deltaDelegatorStartTime.Add(delegationPeriod).Unix()),
+					Wght:   weight,
+				}},
+				pWallet.AVAXAssetID(),
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{deltaDelegationRewardKey.Address()},
+				},
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("stopping beta node to prevent it and its delegator from receiving a validation reward")
+		require.NoError(betaNode.Stop())
+
+		ginkgo.By("waiting until all validation periods are over")
+		// The beta validator was the last added and so has the latest end time. The
+		// delegation periods are shorter than the validation periods.
```

Comment:
`delegation periods are shorter than the validation periods.` --> is this true?
Validator is set to 1 second and the durations are expressed in seconds. Shorter duration would mean delegation has zero duration?